File: app/src/main/java/com/projeto/testedevandroidsantander/ui/mainScreen/MainConfigurator.java
package com.projeto.testedevandroidsantander.ui.mainScreen;

import java.lang.ref.WeakReference;

public enum MainConfigurator {
    INSTANCE;
    public void configure(MainActivity activity){

        MainRouter router = new MainRouter();
        router.activity = new WeakReference<>(activity);

        MainPresenter presenter = new MainPresenter();
        presenter.output = new WeakReference<MainActivityInput>(activity);

        MainInteractor interactor = new MainInteractor();
        interactor.output = presenter;

        LancamentoWorker lancamentoWorker = new LancamentoWorker();
        lancamentoWorker.interactor = interactor;
        interactor.lancamentoWorkerInput = lancamentoWorker;

        if (activity.output == null){
            activity.output = interactor;
        }
        if (activity.router == null){
            activity.router = router;
        }
    }
}

File: Bank/app/src/main/java/com/tata/bank/main/MainConfigurator.kt
package com.tata.bank.main

import java.lang.ref.WeakReference

enum class MainConfigurator {

    INSTANCE;

    fun configure(activity: MainActivity) {
        val router = MainRouter()
        router.activity = WeakReference(activity)

        val presenter = MainPresenter()
        presenter.output = WeakReference(activity)
        presenter.router = router

        val interactor = MainInteractor()
        interactor.output = presenter

        activity.output = interactor
    }
}